### PR TITLE
support lvm/lvms on 4.11/4.12 respectively

### PR DESCRIPTION
**Pull request by subhamkrai**
Signed-off-by: subhamkrai <srai@redhat.com>

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/operators/lvm/config.go [hunk 1]
```diff
@@ -7,13 +7,14 @@ import (
 const (
 	// LvmMinOpenshiftVersion is the minimum OCP version in which lvmo is supported
 	// Any changes here should be updated at line 16 too.
-	LvmMinOpenshiftVersion string = "4.12.0-0.0"
+	LvmMinOpenshiftVersion  string = "4.11.0"
```
Comment (gamli75) [inline internal/operators/lvm/config.go:10]: please change it to LvmoMinOpenshiftVersion

File: internal/operators/lvm/config.go [hunk 2]
```diff
@@ -7,13 +7,14 @@ import (
...
+	lvmsMinOpenshiftVersion string = "4.12.0"
 )
 
 type Config struct {
 	LvmCPUPerHost          int64  `envconfig:"LVM_CPU_PER_HOST" default:"1"`
 	LvmMemoryPerHostMiB    int64  `envconfig:"LVM_MEMORY_PER_HOST_MIB" default:"1200"`
-	LvmMinOpenshiftVersion string `envconfig:"LVM_MIN_OPENSHIFT_VERSION" default:"4.12.0-0.0"`
+	LvmMinOpenshiftVersion string `envconfig:"LVM_MIN_OPENSHIFT_VERSION" default:"4.11.0"`
```
Comment (gamli75) [inline internal/operators/lvm/config.go:17]: Why we need LvmMinOpenshiftVersion twice - here and in the const?

File: internal/operators/lvm/manifest.go [hunk 3]
```diff
@@ -32,11 +35,33 @@ func Manifests() (map[string][]byte, []byte, error) {
...
 
-func getSubscription() ([]byte, error) {
+func getSubscription(cluster *common.Cluster) ([]byte, error) {
+
+	ocpVersion, err := version.NewVersion(cluster.OpenshiftVersion)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	minOCPVersionForLVM, err := version.NewVersion(lvmsMinOpenshiftVersion)
```
Comment (gamli75) [inline internal/operators/lvm/manifest.go:45]:
```suggestion
	minOCPVersionForLVMS, err := version.NewVersion(lvmsMinOpenshiftVersion)
```

File: internal/operators/lvm/manifest.go [hunk 4]
```diff
@@ -32,11 +35,33 @@ func Manifests() (map[string][]byte, []byte, error) {
...
+	if err != nil {
+		return []byte{}, err
+	}
+
+	if ocpVersion.LessThan(minOCPVersionForLVM) {
+		data := map[string]string{
+			"OPERATOR_NAMESPACE":              Operator.Namespace,
+			"OPERATOR_SUBSCRIPTION_NAME":      "odf-lvm-operator",
```
Comment (gamli75) [inline internal/operators/lvm/manifest.go:53]: please use const instead of "odf-lvm-operator"

File: internal/operators/lvm/manifest.go [hunk 5]
```diff
@@ -32,11 +35,33 @@ func Manifests() (map[string][]byte, []byte, error) {
...
+			"OPERATOR_SUBSCRIPTION_SPEC_NAME": "odf-lvm-operator",
+		}
+		return executeTemplate(data, "LvmSubscription", LvmSubscription)
+	}
+
 	data := map[string]string{
-		"OPERATOR_NAMESPACE":         Operator.Namespace,
-		"OPERATOR_SUBSCRIPTION_NAME": Operator.SubscriptionName,
+		"OPERATOR_NAMESPACE":              Operator.Namespace,
+		"OPERATOR_SUBSCRIPTION_NAME":      "lvms-operator",
```
Comment (gamli75) [inline internal/operators/lvm/manifest.go:61]: please use const instead of "lvms-operator"

File: internal/operators/lvm/manifest.go [hunk 6]
```diff
@@ -32,11 +35,33 @@ func Manifests() (map[string][]byte, []byte, error) {
...
+	if err != nil {
+		return []byte{}, err
+	}
+
+	minOCPVersionForLVM, err := version.NewVersion(lvmsMinOpenshiftVersion)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	if ocpVersion.LessThan(minOCPVersionForLVM) {
```
Comment (gamli75) [inline internal/operators/lvm/manifest.go:50]:
```suggestion
        operatorSubscriptionName := LVMS_SUBSCRIPTION_NAME
        operatorSubscriptionSpecName := LVMS_SUBSCRIPTION_SPEC_NAME
	if ocpVersion.LessThan(minOCPVersionForLVMS) {
	         operatorSubscriptionName := LVMO_SUBSCRIPTION_NAME
                 operatorSubscriptionSpecName := LVMO_SUBSCRIPTION_SPEC_NAME
                 }
                 data := map[string]string{
			"OPERATOR_NAMESPACE":              Operator.Namespace,
			"OPERATOR_SUBSCRIPTION_NAME":     operatorSubscriptionName,
                        "OPERATOR_SUBSCRIPTION_SPEC_NAME": operatorSubscriptionSpecName,
		}
		return executeTemplate(data, "LvmSubscription", LvmSubscription)
	}
```

File: internal/operators/lvm/config.go (hunk 1 above)
Comment (subhamkrai) [reply internal/operators/lvm/config.go]: done

File: internal/operators/lvm/config.go (hunk 2 above)
Comment (subhamkrai) [reply internal/operators/lvm/config.go:20]: we use this const in other pkg like cnv, and we need to get min openshift for lvmo/s support and there will case when this struct is not filled with default values and we are trying to get the  min openshift for lvmo/s.

File: internal/operators/lvm/manifest.go (hunk 3 above)
Comment (subhamkrai) [reply internal/operators/lvm/manifest.go]: done

File: internal/operators/lvm/manifest.go (hunk 4 above)
Comment (subhamkrai) [reply internal/operators/lvm/manifest.go]: declare const in one place, config file

File: internal/operators/lvm/manifest.go (hunk 5 above)
Comment (subhamkrai) [reply internal/operators/lvm/manifest.go]: declare const in one place, config file

File: internal/operators/lvm/manifest.go (hunk 6 above)
Comment (subhamkrai) [reply internal/operators/lvm/manifest.go]: declared const in one place, config file

File: internal/operators/cnv/cnv_operator.go [hunk 7]
```diff
@@ -68,7 +68,7 @@ func (o *operator) GetDependencies(cluster *common.Cluster) ([]string, error) {
 		return []string{}, err
 	}
 
-	minOCPVersionForLVM, err := version.NewVersion(lvm.LvmMinOpenshiftVersion)
+	minOCPVersionForLVM, err := version.NewVersion(lvm.LvmoMinOpenshiftVersion)
```
Comment (gamli75) [inline internal/operators/cnv/cnv_operator.go:71]:
```suggestion
	minOCPVersionForLVMS, err := version.NewVersion(lvm.LvmsMinOpenshiftVersion)
```

Comment (gamli75) [reply internal/operators/cnv/cnv_operator.go:71]: in CNV - we want to install LVMS only when OCP version >= 4.12, in 4.11 we use LSO

File: internal/operators/manager.go [hunk 8]
```diff
@@ -403,7 +403,7 @@ func (mgr *Manager) GetSupportedOperatorsByType(operatorType models.OperatorType
 
 func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string, operators []*models.MonitoredOperator) error {
 
-	minOCPVersionForLVM, err := version.NewVersion(lvm.LvmMinOpenshiftVersion)
+	minOCPVersionForLVM, err := version.NewVersion(lvm.LvmoMinOpenshiftVersion)
```
Comment (gamli75) [inline internal/operators/manager.go:406]:
```suggestion
	minOCPVersionForLVMO, err := version.NewVersion(lvm.LvmoMinOpenshiftVersion)
```

File: internal/operators/lvm/manifest.go [hunk 9]
```diff
@@ -32,11 +35,33 @@ func Manifests() (map[string][]byte, []byte, error) {
...
+			"OPERATOR_SUBSCRIPTION_SPEC_NAME": lvmoSubscriptionName,
+		}
+		return executeTemplate(data, "LvmSubscription", LvmSubscription)
+	}
+
 	data := map[string]string{
-		"OPERATOR_NAMESPACE":         Operator.Namespace,
-		"OPERATOR_SUBSCRIPTION_NAME": Operator.SubscriptionName,
+		"OPERATOR_NAMESPACE":              Operator.Namespace,
+		"OPERATOR_SUBSCRIPTION_NAME":      lvmsMinOpenshiftVersion,
```
Comment (gamli75) [inline internal/operators/lvm/manifest.go:61]:
```suggestion
		"OPERATOR_SUBSCRIPTION_NAME":      lvmsSubscriptionName,
		"OPERATOR_SUBSCRIPTION_SPEC_NAME": lvmsSubscriptionName,
```

File: internal/operators/cnv/cnv_operator.go (hunk 7 above)
Comment (subhamkrai) [reply internal/operators/cnv/cnv_operator.go]: got it

File: internal/operators/lvm/manifest.go (hunk 9 above)
Comment (subhamkrai) [reply internal/operators/lvm/manifest.go]: done

File: internal/operators/manager.go (hunk 8 above)
Comment (subhamkrai) [reply internal/operators/manager.go]: done

**Comment by gamli75** [comment]
replaced by https://github.com/openshift/assisted-service/pull/4930

**Pull request closed without merging**